drivers/led/lp5521.c

```diff
@@ -0,0 +1,921 @@
+{
+	const struct lp5521_config *config = dev->config;
+	int ret;
+	uint8_t tx_buf[4] = {
+		LP5521_R_CURRENT,
+		config->r_current,
+		config->g_current,
+		config->b_current
+		};
+
+	ret = i2c_write_dt(&config->bus, tx_buf, sizeof(tx_buf));
+
+	return ret;
+}
+
+static int lp5521_led_init(const struct device *dev)
+{
+	const struct lp5521_config *config = dev->config;
+	struct lp5521_data *data = dev->data;
+	struct led_data *dev_data = &data->dev_data;
+	const struct gpio_dt_spec *en_pin = &config->en_pin;
+
+	int ret;
+
+	if (en_pin != 0) {
+		if (!device_is_ready(en_pin->port)) {
+			return -ENODEV;
+		}
+		if (gpio_pin_configure_dt(en_pin, GPIO_OUTPUT_ACTIVE)) {
+			return -ENODEV;
+		}
+		if (gpio_pin_set_dt(en_pin, true)) {
+			return -ENODEV;
+		}
+		k_msleep(5);
+	}
+
+	if (!device_is_ready(config->bus.bus)) {
+		LOG_ERR("I2C device not ready");
+		return -ENODEV;
+	}
+
+	/* Hardware specific limits */
+	dev_data->min_period = LP5521_MIN_BLINK_PERIOD;
+	dev_data->max_period = LP5521_MAX_BLINK_PERIOD;
+	dev_data->min_brightness = LP5521_MIN_BRIGHTNESS;
+	dev_data->max_brightness = LP5521_MAX_BRIGHTNESS;
+
+	ret = lp5521_led_update_current(dev);
+	if (ret) {
+		LOG_ERR("Setting current setting LP5521 LED chip failed.");
+		return ret;
+	}
+
+	if (i2c_reg_write_byte_dt(&config->bus, LP5521_ENABLE,
+					LP5521_ENABLE_CHIP_EN)) {
+		LOG_ERR("Enabling LP5521 LED chip failed.");
+		return -EIO;
+	}
+
+	if (i2c_reg_write_byte_dt(&config->bus, LP5521_CONFIG, 0x11)) {
+		LOG_ERR("Configuring LP5521 LED chip failed.");
+		return -EIO;
+	}
+
+	if (i2c_reg_write_byte_dt(&config->bus, LP5521_OP_MODE, 0x3F)) {
+		LOG_ERR("Disabling all engines failed.");
+		return -EIO;
+	}
+
+	return 0;
+}
+
+static const struct led_driver_api lp5521_led_api = {
+	.blink = lp5521_led_blink,
+	.set_brightness = lp5521_led_set_brightness,
+	.on = lp5521_led_on,
+	.off = lp5521_led_off,
+};
+
+#define LP5521_DEFINE(id)				\
+	BUILD_ASSERT(DT_INST_PROP(id, red_output_current) <= \
+					LP5521_MAX_CURRENT_SETTING,			\
+	"Red channel current must be between 0 and 25.5 mA.");\
+	BUILD_ASSERT(DT_INST_PROP(id, green_output_current) <=\
+					LP5521_MAX_CURRENT_SETTING,			\
+	"Green channel current must be between 0 and 25.5 mA.");\
+	BUILD_ASSERT(DT_INST_PROP(id, blue_output_current) <=\
+					LP5521_MAX_CURRENT_SETTING,		\
+	"Blue channel current must be between 0 and 25.5 mA.");\
+	\
+	static const struct lp5521_config lp5521_config_##id = {\
+		.bus = I2C_DT_SPEC_INST_GET(id),					\
+		.r_current = DT_INST_PROP(id, red_output_current),	\
+		.g_current = DT_INST_PROP(id, green_output_current),\
+		.b_current = DT_INST_PROP(id, blue_output_current),	\
+		.en_pin = GPIO_DT_SPEC_INST_GET_OR(id, en_gpios,\
+											 {0}),\
+	};	\
+	\
+	struct lp5521_data lp5521_data_##id;			\
+	DEVICE_DT_INST_DEFINE(id, &lp5521_led_init, \
+			NULL,					\
+			&lp5521_data_##id,		\
+			&lp5521_config_##id, POST_KERNEL,\
+			CONFIG_LED_INIT_PRIORITY,	\
+			&lp5521_led_api);
```

Comment:
This is not addressed. Please format back-slashes
```suggestion
#define LP5521_DEFINE(id)							\
	BUILD_ASSERT(DT_INST_PROP(id, red_output_current) <=			\
						LP5521_MAX_CURRENT_SETTING,	\
		"Red channel current must be between 0 and 25.5 mA.");		\
	BUILD_ASSERT(DT_INST_PROP(id, green_output_current) <=			\
						LP5521_MAX_CURRENT_SETTING,	\
		"Green channel current must be between 0 and 25.5 mA.");	\
	BUILD_ASSERT(DT_INST_PROP(id, blue_output_current) <=			\
						LP5521_MAX_CURRENT_SETTING,	\
		"Blue channel current must be between 0 and 25.5 mA.");		\
										\
	static const struct lp5521_config lp5521_config_##id = {		\
		.bus = I2C_DT_SPEC_INST_GET(id),				\
		.r_current = DT_INST_PROP(id, red_output_current),		\
		.g_current = DT_INST_PROP(id, green_output_current),		\
		.b_current = DT_INST_PROP(id, blue_output_current),		\
		.en_pin = GPIO_DT_SPEC_INST_GET_OR(id, en_gpios, {0}),		\
	};									\
										\
	struct lp5521_data lp5521_data_##id;					\
	DEVICE_DT_INST_DEFINE(id, &lp5521_led_init, NULL,			\
			&lp5521_data_##id,					\
			&lp5521_config_##id, POST_KERNEL,			\
			CONFIG_LED_INIT_PRIORITY,				\
			&lp5521_led_api);

```